parts/k8s/kubernetesmastercustomdata.yml

```diff
@@ -512,6 +518,10 @@ runcmd:
 {{if .OrchestratorProfile.KubernetesConfig.EnableAggregatedAPIs}}
 - sudo bash /etc/kubernetes/generate-proxy-certs.sh
 {{end}}
+- /bin/echo DAEMON_ARGS=--name "{{WrapAsVerbatim "variables('masterVMNames')[copyIndex(variables('masterOffset'))]"}}" --peer-client-cert-auth --peer-trusted-ca-file={{WrapAsVariable "etcdCaFilepath"}} --peer-cert-file={{WrapAsVerbatim "variables('etcdPeerCertFilepath')[copyIndex(variables('masterOffset'))]"}} --peer-key-file={{WrapAsVerbatim "variables('etcdPeerKeyFilepath')[copyIndex(variables('masterOffset'))]"}} --initial-advertise-peer-urls "{{WrapAsVerbatim "variables('masterEtcdTLSPeerURLs')[copyIndex(variables('masterOffset'))]"}}" --listen-peer-urls "{{WrapAsVerbatim "variables('masterEtcdTLSPeerURLs')[copyIndex(variables('masterOffset'))]"}}" --client-cert-auth --trusted-ca-file={{WrapAsVariable "etcdCaFilepath"}} --cert-file={{WrapAsVariable "etcdServerCertFilepath"}} --key-file={{WrapAsVariable "etcdServerKeyFilepath"}} --advertise-client-urls "{{WrapAsVerbatim "variables('masterEtcdTLSClientURLs')[copyIndex(variables('masterOffset'))]"}}" --listen-client-urls "{{WrapAsVerbatim "concat(variables('masterEtcdTLSClientURLs')[copyIndex(variables('masterOffset'))], ',https://127.0.0.1:', variables('masterEtcdClientPort'))"}}" --initial-cluster-token "k8s-etcd-cluster" --initial-cluster "{{WrapAsVerbatim "variables('masterEtcdTLSClusterStates')[div(variables('masterCount'), 2)]"}} --data-dir "/var/lib/etcddisk"" --initial-cluster-state "new" > /tmp/etcdtls
```

Comment:
For the peer communication part:
do we need the --discovery ${DISCOVERY_URL} argument, per https://coreos.com/etcd/docs/latest/op-guide/security.html#example-3-transport-security--client-certificates-in-a-cluster.

For the client-to-server part:
--advertise-client-urls does not have 127.0.0.1 and port setup as --listen-client-urls, is that intentional?
--advertise-client-urls "{{WrapAsVerbatim "variables('masterEtcdTLSClientURLs')[copyIndex(variables('masterOffset'))]"}}" --listen-client-urls "{{WrapAsVerbatim "concat(variables('masterEtcdTLSClientURLs')[copyIndex(variables('masterOffset'))], ',https://127.0.0.1:', variables('masterEtcdClientPort'))"}}"

Reply:
Based on this https://coreos.com/etcd/docs/latest/op-guide/runtime-reconf-design.html#do-not-use-public-discovery-service-for-runtime-reconfiguration I assumed we didn't since this is reconfiguring a cluster.

Reply:
I followed the original config of the cluster for --advertise-client-urls and --listen-client-urls which was as follows:
--advertise-client-urls "{{WrapAsVerbatim "variables('masterEtcdClientURLs')[copyIndex(variables('masterOffset'))]"}}" --listen-client-urls "{{WrapAsVerbatim "concat(variables('masterEtcdClientURLs')[copyIndex(variables('masterOffset'))], ',http://127.0.0.1:', variables('masterEtcdClientPort'))"}}"

https://github.com/Azure/acs-engine/blob/709e301160210b29f09d036f6b2d28c4778d823f/parts/k8s/kubernetesmastercustomdata.yml#L489

Reply:
I see, thanks for the explanation!